src/parsers/locale-messagesjson.js

```diff
@@ -153,6 +166,23 @@ export default class LocaleMessagesJSONParser extends JSONParser {
         );
       }
 
+      if (messageNameMatch) {
+        const messageName = messageNameMatch[1];
+        let nameContainsInvalidWords = false;
+        if (message === messageName) {
+          const nameLowerCase = this.parsedJSON[message].message.toLowerCase();
+          nameContainsInvalidWords = NOT_ALLOWED_NAME_WORDS.some((word) =>
+            nameLowerCase.includes(word)
+          );
+        }
+        if (nameContainsInvalidWords) {
+          this.collector.addWarning(
+            messages.PROP_NAME_MUST_NOT_CONTAIN_MOZILLA_OR_FIREFOX
+          );
```

Comment:
You probably want to override the `file` here, since the default is `MANIFEST_JSON`, but we are actually in a locale file. The following should work:

```suggestion
          this.collector.addWarning({
            ...messages.PROP_NAME_MUST_NOT_CONTAIN_MOZILLA_OR_FIREFOX,
            file: this.filename,
          });
```